ddtrace/settings/config.py

```diff
@@ -187,6 +196,72 @@ def get_error_ranges(error_range_str):
     return error_ranges  # type: ignore[return-value]
 
 
+ConfigSource = Literal["default", "env", "code"]
+
+
+class _ConfigItem:
+    """Configuration item that tracks the value of a setting, and where it came from."""
+
+    def __init__(self, name, default, envs):
+        self._name = name
+        self._default_value = default
+        self._env_value = None
+        self._code_value = None
+        self._envs = envs
+        for (env_var, parser) in envs:
+            if env_var in os.environ:
+                self._env_value = parser(os.environ[env_var])
+                break
+
+    def set_code(self, value):
+        self._code_value = value
+
+    def value(self):
+        if self._code_value is not None:
+            return self._code_value
+        if self._env_value is not None:
+            return self._env_value
+        return self._default_value
+
+    def source(self):
+        # type: () -> ConfigSource
+        if self._code_value is not None:
+            return "code"
+        if self._env_value is not None:
+            return "env"
+        return "default"
+
+    def __repr__(self):
+        return "<{} name={} default={} env_value={} user_value={}>".format(
+            self.__class__.__name__,
+            self._name,
+            self._default_value,
+            self._env_value,
+            self._code_value,
+        )
+
+
+def _default_config():
+    # type: () -> Dict[str, _ConfigItem]
+    return {
+        "_trace_sample_rate": _ConfigItem(
+            name="trace_sample_rate",
```

Comment:
I like this convention. Configuration names should not be prefixed with an underscore. Let's stick to this 😤